### PR TITLE
Added attributes pc_tag and scope to fvCtx

### LINKED ISSUE #1237: Add two new attributes pc_tag and scope to fvCtx
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s) + ACI Class(es):

* aci_vrf + fv:Ctx

### APIC version and APIC Platform

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by abrahammughal**
Fixes #1237

### CODE REVIEW

File: aci/resource_aci_fvctx.go [hunk 1]
```diff
@@ -213,6 +225,8 @@ func setVRFAttributes(fvCtx *models.VRF, d *schema.ResourceData) (*schema.Resour
 	d.Set("name", fvCtxMap["name"])
 
 	d.Set("annotation", fvCtxMap["annotation"])
+	d.Set("scope", fvCtxMap["scope"])
```
Comment (akinross) [inline aci/resource_aci_fvctx.go:228]: did you test adding/updating these values?

**Review by akinross — CHANGES_REQUESTED** [review]

File: aci/resource_aci_fvctx.go (hunk 1 above)
Comment (abrahammughal) [reply aci/resource_aci_fvctx.go:228]: Yes, I did

Comment (akinross) [reply aci/resource_aci_fvctx.go:228]: How did you test this behaviour? 

I think this only reads the data from apic and does not set it in the payload.

I would expect you to set it [here](https://github.com/CiscoDevNet/terraform-provider-aci/pull/1238/files#diff-0baf6466d84e03887920db87e9afed21095c499dd9a4c58bc6cff26d77dc05f1R274-R298) and that fvCtxAttr it is passed into the `models.NewVRF(fmt.Sprintf("ctx-%s", name), TenantDn, desc, fvCtxAttr)` function with the scope and pc_tag attribute.

The same would apply for the update.

Also I do not see update of the aci-go-client vendor files.

File: vendor/github.com/ciscoecosystem/aci-go-client/v2/models/fv_ctx.go [hunk 2]
```diff
@@ -19,6 +19,10 @@ type VRFAttributes struct {
 
 	Annotation string `json:",omitempty"`
 
+	PcTag string `json:",omitempty"`
```
Comment (akinross) [inline vendor/github.com/ciscoecosystem/aci-go-client/v2/models/fv_ctx.go:22]: How did you add the vendor file?

**Review by akinross — CHANGES_REQUESTED** [review]

File: vendor/github.com/ciscoecosystem/aci-go-client/v2/models/fv_ctx.go (hunk 2 above)
Comment (abrahammughal) [reply vendor/github.com/ciscoecosystem/aci-go-client/v2/models/fv_ctx.go:22]: Just ran go mod vendor

File: aci/data_source_aci_fvctx.go [hunk 3]
```diff
@@ -27,6 +27,18 @@ func dataSourceAciVRF() *schema.Resource {
 				Required: true,
 			},
 
+			"pc_tag": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (akinross) [inline aci/data_source_aci_fvctx.go]: remove optional

File: aci/data_source_aci_fvctx.go [hunk 4]
```diff
@@ -27,6 +27,18 @@ func dataSourceAciVRF() *schema.Resource {
...
+				Computed: true,
+			},
+
+			"scope": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (akinross) [inline aci/data_source_aci_fvctx.go]: remove optional

File: aci/resource_aci_fvctx.go [hunk 5]
```diff
@@ -38,6 +38,18 @@ func resourceAciVRF() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"pc_tag": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (akinross) [inline aci/resource_aci_fvctx.go]: remove optional

File: aci/resource_aci_fvctx.go [hunk 6]
```diff
@@ -38,6 +38,18 @@ func resourceAciVRF() *schema.Resource {
...
+				Computed: true,
+			},
+
+			"scope": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (akinross) [inline aci/resource_aci_fvctx.go]: remove optional

File: docs/data-sources/vrf.md [hunk 7]
```diff
@@ -29,6 +29,8 @@ data "aci_vrf" "dev_ctx" {
 
 - `id` - Attribute id set to the Dn of the VRF.
 - `annotation` - (Optional) Annotation tags for object VRF.
+- `pc_tag` - (Optional) A numeric ID to represent an EPG.
+- `scope` - (Optional) A numeric value for other APICs to identify VRF.
```
Comment (akinross) [inline docs/data-sources/vrf.md:33]:
```suggestion
- `pc_tag` - (Read-Only) A numeric ID to represent an EPG.
- `scope` - (Read-Only) A numeric value for other APICs to identify VRF.
```

File: legacy-docs/docs/d/vrf.html.markdown [hunk 8]
```diff
@@ -29,6 +29,8 @@ data "aci_vrf" "dev_ctx" {
 
 - `id` - Attribute id set to the Dn of the VRF.
 - `annotation` - (Optional) Annotation tags for object VRF.
+- `pc_tag` - (Optional) A numeric ID to represent an EPG.
+- `scope` - (Optional) A numeric value for other APICs to identify VRF.
```
Comment (akinross) [inline legacy-docs/docs/d/vrf.html.markdown:33]:
```suggestion
- `pc_tag` - (Read-Only) A numeric ID to represent an EPG.
- `scope` - (Read-Only) A numeric value for other APICs to identify VRF.
```

File: legacy-docs/docs/r/vrf.html.markdown [hunk 9]
```diff
@@ -46,6 +46,8 @@ resource "aci_vrf" "foovrf" {
 - `tenant_dn` - (Required) Distinguished name of parent Tenant object.
 - `name` - (Required) Name of Object VRF.
 - `annotation` - (Optional) Annotation tags for object VRF.
+- `pc_tag` - (Optional) A numeric ID to represent an EPG.
+- `scope` - (Optional) A numeric value for other APICs to identify VRF.
```
Comment (akinross) [inline legacy-docs/docs/r/vrf.html.markdown:50]:
```suggestion
- `pc_tag` - (Optional) A numeric ID to represent an EPG.
- `scope` - (Optional) A numeric value for other APICs to identify VRF.
```
```suggestion
- `pc_tag` - (Read-Only) A numeric ID to represent an EPG.
- `scope` - (Read-Only) A numeric value for other APICs to identify VRF.
```

**Review by akinross — CHANGES_REQUESTED** [review]

File: aci/resource_aci_fvctx.go (hunk 1 above)
Comment (akinross) [reply aci/resource_aci_fvctx.go:226]: create and update not required for these attributes

**Review by akinross — DISMISSED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by gmicol — DISMISSED** [review]
LGTM

File: docs/data-sources/vrf.md [hunk 10]
```diff
@@ -29,6 +29,8 @@ data "aci_vrf" "dev_ctx" {
 
 - `id` - Attribute id set to the Dn of the VRF.
 - `annotation` - (Optional) Annotation tags for object VRF.
+- `pc_tag` - (Read-Only) A numeric ID to represent an EPG.
```
Comment (lhercot) [inline docs/data-sources/vrf.md:32]:
```suggestion
- `pc_tag` - (Read-Only) A numeric ID to represent a policy enforcement group.
```

File: docs/data-sources/vrf.md [hunk 11]
```diff
@@ -29,6 +29,8 @@ data "aci_vrf" "dev_ctx" {
...
+- `scope` - (Read-Only) A numeric value for other APICs to identify VRF.
```
Comment (lhercot) [inline docs/data-sources/vrf.md:33]:
```suggestion
- `scope` - (Read-Only) A numeric value for other APICs to identify a specific VRF.
```

File: docs/data-sources/vrf.md (hunk 10 above)
Comment (lhercot) [reply docs/data-sources/vrf.md:32]: change it for resource too

File: docs/data-sources/vrf.md (hunk 11 above)
Comment (lhercot) [reply docs/data-sources/vrf.md:33]: change it for resource too

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by samiib — APPROVED** [review]
LGTM

**Review by gmicol — APPROVED** [review]
LGTM

**Review by akinross — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**